addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -347,53 +250,74 @@ def _prepare_down_payment_lines_values(self, order):
                 down_payment_values.append([
                     pct_tax,
                     line['analytic_distribution'],
-                    line['quantity'] * fixed_tax.amount
+                    line['quantity'] * fixed_tax.amount,
+                    account
                 ])
 
         downpayment_line_map = {}
-        for tax_id, analytic_distribution, price_subtotal in down_payment_values:
+        for tax_id, analytic_distribution, price_subtotal, account in down_payment_values:
             grouping_key = frozendict({
                 'tax_id': tuple(sorted(tax_id.ids)),
                 'analytic_distribution': analytic_distribution,
+                'account_id': account,
             })
             downpayment_line_map.setdefault(grouping_key, {
                 **base_downpayment_lines_values,
-                **grouping_key,
+                'tax_id': grouping_key['tax_id'],
+                'analytic_distribution': grouping_key['analytic_distribution'],
                 'product_uom_qty': 0.0,
                 'price_unit': 0.0,
             })
-            downpayment_line_map[grouping_key]['price_unit'] += \
-                order.currency_id.round(price_subtotal * percentage)
+            downpayment_line_map[grouping_key]['price_unit'] += price_subtotal * ratio
+
+        res = [
+            list(downpayment_line_map.values()),
+            [key['account_id'].id for key in downpayment_line_map]
+        ]
+        return res
```

Comment:
I wonder if it would not make returning the dictionary instead make it easier also for the analytic distribution